alphagov/paas-bootstrap
# [#131876783] Manage datadog secrets in one script

[#131876783 Migrate paas-cf to paas-bootstrap](https://www.pivotaltracker.com/story/show/131876783)

What?
----

Add logic to allow manage datadog secrets during bootstrap and from the pipeline using S3 buckets.

See commits for details.

Note: I personally consider this problem could be solved by using a centralised Key-Value store independent of the platform, like credstash or so. But that implementation is out of the scope.

How to review?
-------------

 * Check the code makes sense.
 * Remove the datadog secrets from the state bucket `aws s3 rm s3://gds-paas-${DEPLOY_ENV}-state/datadog-secrets.yml`
    * Try to apply the pipeline with datadog, the credentials should be set from paas-pass
    * Upload the datadog file with `make dev upload-datadog-secrets DEPLOY_ENV=...`
    * Apply the secrets again, the credentials would be loaded from the bucket.

Who?
----

Anyone but @keymon or @alext
